dispatcher/MongoConnect.py

```diff
@@ -261,66 +269,122 @@ def aggregate_status(self):
             return None
 
-    def get_configured_nodes(self, detector, link_mv, link_nv):
+    def is_linked(self, a, b):
+        '''
+        Check if the detectors are in a compatible linked configuration.
         '''
-        Get the nodes we want from the config file
+        mode_a = self.goal_state[a]["mode"]
+        mode_b = self.goal_state[b]["mode"]
+        doc_a = self.collections['options'].find_one({'name': mode_a})
+        doc_b = self.collections['options'].find_one({'name': mode_b})
+        detectors_a = doc_a['detector']
+        detectors_b = doc_b['detector']
+
+        # Check if the linked detectors share the same run mode and
+        # if they are both present in the detectors list of that mode
+        return mode_a == mode_b and a in detectors_b and b in detectors_a
+
+    def get_super_detector(self):
         '''
-        retnodes = []
-        retcc = []
-        retnodes = list(self.latest_status[detector]['readers'].keys())
-        retcc = list(self.latest_status[detector]['controller'].keys())
-        if detector == 'tpc' and link_nv == 'true':
-            retnodes += list(self.latest_status['neutron_veto']['readers'].keys())
-            retcc += list(self.latest_status['neutron_veto']['controllers'].keys())
-        if detector == 'tpc' and link_mv == 'true':
-            retnodes += list(self.latest_status['muon_veto']['readers'].keys())
-            retcc += list(self.latest_status['muon_veto']['controllers'].keys())
-        return retnodes, retcc
+        Get the Super Detector configuration
+        if the detectors are in a compatible linked mode.
+        - case A: tpc, mv and nv all linked
+        - case B: tpc, mv and nv all un-linked
+        - case C: tpc and mv linked, nv un-linked
+        - case D: tpc and nv linked, mv un-linked
+        - case E: tpc unlinked, mv and nv linked
+        We will check the compatibility of the linked mode for a pair of detectors per time.
+        '''
+        ret = {'tpc': {'controller': list(self.dc['tpc']['controller'].keys()),
+                       'readers': list(self.dc['tpc']['readers'].keys()),
+                       'detectors': ['tpc']}}
+        mv = self.dc['muon_veto']
+        nv = self.dc['neutron_veto']
+
+        # tpc and muon_veto linked mode
+        if self.is_linked('tpc', 'muon_veto'):
+            # case A or C
+            ret['tpc']['controller'] += list(mv['controller'].keys())
+            ret['tpc']['readers'] += list(mv['readers'].keys())
+            ret['tpc']['detectors'] += ['muon_veto']
+        else:
+            # case B or E
+            ret['muon_veto'] = {'controller': list(mv['controller'].keys()),
+                    'readers': list(mv['readers'].keys()),
+                    'detectors': ['muon_veto']}
+        if self.is_linked('tpc', 'neutron_veto'):
+            # case A or D
+            ret['tpc']['controller'] += list(nv['controller'].keys())
```

Comment:
Especially, having multiple ccs in the config is not supported properly in the dispatcher (see comment below)

Reply:
It actually is